### PR TITLE
add pipenv file

### PR DESCRIPTION
**Pull request by Trafitto**
I hope it goes well, I had never used pipenv #5

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+[packages]
+numpy = "*"
+opencv-python = "*"
+streamlit = "*"
+pandas = "*"
+altair = "*"
+
+[requires]
+python_version = "3.6"
```
Comment (treuille) [inline Pipfile]: Python 2.7 is also supported.

File: Pipfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+url = "https://pypi.org/simple"
+verify_ssl = true
+
+[dev-packages]
+
+[packages]
+numpy = "*"
+opencv-python = "*"
+streamlit = "*"
+pandas = "*"
```
Comment (treuille) [inline Pipfile]: Pandas is imported by Streamlit.

File: Pipfile [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+name = "pypi"
+url = "https://pypi.org/simple"
+verify_ssl = true
+
+[dev-packages]
+
+[packages]
+numpy = "*"
+opencv-python = "*"
+streamlit = "*"
```
Comment (treuille) [inline Pipfile]: Streamlit version must be at least 0.47.4

File: Pipfile [hunk 4]
```diff
@@ -0,0 +1,16 @@
+[[source]]
```
Comment (treuille) [inline Pipfile]: This repo hasn't been uploaded to pypi. Should this be here?

File: Pipfile.lock [hunk 5]
```diff
@@ -0,0 +1,447 @@
+{
```
Comment (treuille) [inline Pipfile.lock]: I don't think a Pipfile lock should be checked into the repo.

File: Pipfile [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+name = "pypi"
+url = "https://pypi.org/simple"
+verify_ssl = true
+
+[dev-packages]
+
+[packages]
+numpy = "*"
```
Comment (monchier) [inline Pipfile]: numpy also is provided by Streamlit

File: Pipfile [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+opencv-python = "*"
+streamlit = "*"
+pandas = "*"
+altair = "*"
```
Comment (monchier) [inline Pipfile]: `altair` also is provided by Streamlit

File: .gitignore [hunk 8]
```diff
@@ -3,3 +3,4 @@ __pycache__/
 *$py.class
 yolov3.weights
 yolov3.cfg
+.vscode
```
Comment (monchier) [inline .gitignore]: I would rather keep this IDE-independent. Can you remove?

**Comment by Trafitto** [comment]
Ok, I make the requested changes

File: Pipfile (hunk 1 above)
Comment (Trafitto) [reply Pipfile]: I think it's better to use python 3*

https://github.com/pypa/pipfile/issues/87
https://pythonclock.org/
https://www.python.org/doc/sunset-python-2/

**Comment by Trafitto** [comment]
I made the requested changes.

Here are the steps to create pipenv:
`pipenv shell`
`pipenv install streamlit`
`pipenv opencv-python`
Use `pipenv check` for checking the pipfile
Use `pipenv install` for installing all dependencies

File: .gitignore [hunk 9]
```diff
@@ -2,4 +2,4 @@ __pycache__/
 *.py[cod]
 *$py.class
 yolov3.weights
-yolov3.cfg
+yolov3.cfg
```
Comment (monchier) [inline .gitignore]: I think the original file may have had a newline. We should preserve it.

File: Pipfile (hunk 1 above)
Comment (monchier) [reply Pipfile]: Please, remove this constraint. It is a requirement for us to support Python 2.7 as well.

File: Pipfile [hunk 10]
```diff
@@ -0,0 +1,13 @@
+[[source]]
+name = "pypi"
+url = "https://pypi.org/simple"
+verify_ssl = true
+
+[dev-packages]
+
+[packages]
+streamlit = ">0.47.4"
```
Comment (monchier) [inline Pipfile]: Is not 0.47.0 enough for this to work?

Comment (Trafitto) [reply Pipfile]: Is the version suggested by @treuille 
> Streamlit version must be at least 0.47.4

File: Pipfile (hunk 1 above)
Comment (Trafitto) [reply Pipfile]: Removed!

File: .gitignore (hunk 9 above)
Comment (Trafitto) [reply .gitignore]: I don't think it changes to have an extra line, but I added it. Let me know if there are other changes to make

**Comment by treuille** [comment]
Congratulations on your first merge to the repo @Trafitto!

**Comment by Trafitto** [comment]
Thanks :+1:

**Pull request merged**